bundles/umljava/tools.vitruv.applications.umljava.java2uml/src/tools/vitruv/applications/umljava/java2uml/JavaToUmlClassifier.reactions

```diff
@@ -393,12 +393,26 @@ routine removeUmlSuperInterface(java::Interface jInterface, java::TypeReference
 reaction JavaPackageCreated {
     after element java::Package inserted as root
     call {
+    	createPackageEClassCorrespondence(newValue)
     	detectOrCreateUmlModel(newValue)
 		registerPredefinedUmlPrimitiveTypes(correspondenceModel, newValue.eResource.resourceSet)
 	    createOrFindUmlPackage(newValue)
     }
 }
 
+/**
+ * Required to enable locating existing packages with missing correspondences when keeping more than two models consistent.
+ */
+routine createPackageEClassCorrespondence(java::Package jPackage) {
+	match {
+		val allPackages = retrieve many java::Package corresponding to jPackage.eClass
+		check !allPackages.contains(jPackage)
+	}
+    action {
+        add correspondence between jPackage and jPackage.eClass
+    }
+}
```

Comment:
When creating the same correspondence in different reactions, do I need to check if the correspondence has been already made (like in the referenced code snippet), or is there no such thing as creating a correspondence twice?

Reply:
I am pretty sure that you have to check whether a correspondence already exists, as there is no check whether an equal correspondence already exists. So the current implementation is reasonable.